elide-async/src/main/java/com/yahoo/elide/async/models/AsyncQuery.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.async.models;
+
+import java.security.Principal;
+import java.util.Date;
+import java.util.UUID;
+
+import javax.inject.Inject;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.OneToOne;
+import javax.persistence.PrePersist;
+import javax.persistence.PreUpdate;
+import javax.persistence.Transient;
+
+import com.yahoo.elide.annotation.Include;
+import com.yahoo.elide.annotation.OnCreatePostCommit;
+import com.yahoo.elide.annotation.ReadPermission;
+import com.yahoo.elide.annotation.UpdatePermission;
+import com.yahoo.elide.async.service.AsyncExecutorService;
+import com.yahoo.elide.core.RequestScope;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Model for Async Query
+ */
+@Entity
+@Include(type = "query", rootLevel = true)
+@ReadPermission(expression = "Principal is Owner")
+@UpdatePermission(expression = "Prefab.Role.None")
```

Comment:
What about DeletePermission?

Reply:
Will make the DeletePermission as None .